src/sync/client.rs

```diff
@@ -42,17 +42,21 @@ pub struct Client {
 }
 
 impl Client {
-    /// Initialize a new [`Client`].
-    pub fn new(fd: RawFd) -> Client {
+    pub fn connect_unix(path: &str) -> Result<Client> {
+        let fd = unsafe { client_connect_unix(path)? };
+        Self::new(fd)
+    }
+
+    /// Initialize a new [`Client`] from raw file descriptor.
+    pub fn new(fd: RawFd) -> Result<Client> {
```

Comment:
@mxpv Sorry to notice this so late, After this line changed we must bump the major version, but I think it is not the right time to bump the major version, what do you think? How about split this to a new pr?

Reply:
I've reverted this to keep avoid API backward compatible.